### PR TITLE
refactor: remove `getValue()`

### LINKED ISSUE #463: todo: remove `getValue()`
[`getValue()`](https://github.com/denoland/saaskit/blob/0f61d354a36d4f877c58feb162531d3cf087e5f7/utils/db.ts#L16-L22) is too trivial. Best we delete it and just use `entry.value` where needed.

### PR DESCRIPTION
**Pull request by roberto-morado**
fixes #463

### CODE REVIEW

File: utils/db.ts [hunk 1]
```diff
@@ -128,7 +120,8 @@ export async function deleteItem(item: Item) {
 }
 
 export async function getItem(id: string) {
-  return await getValue<Item>(["items", id]);
+  const itemsByIdKey = ["items", id];
+  return (await kv.get<Item>(itemsByIdKey)).value;
```
Comment (iuioiua) [inline utils/db.ts:124]:
```suggestion
  const res = await kv.get<Item>(["items", id]);
  return res.value;
```
This pattern is used in [the docs](https://deno.land/manual@v1.36.4/runtime/kv). Let's stick to that 🙂

**Review by iuioiua — CHANGES_REQUESTED** [review]
Thanks, @roberto-morado! One main concern.

File: utils/db.ts (hunk 1 above)
Comment (roberto-morado) [reply utils/db.ts:124]: Gotcha

File: utils/db.ts [hunk 2]
```diff
@@ -128,7 +120,9 @@ export async function deleteItem(item: Item) {
 }
 
 export async function getItem(id: string) {
-  return await getValue<Item>(["items", id]);
+  const itemsByIdKey = ["items", id];
+  const res = await kv.get<Item>(itemsByIdKey);
+  return res.value;
```
Comment (iuioiua) [inline utils/db.ts:125]:
```suggestion
  const res = await kv.get<Item>(["items", id]);
  return res.value;
```
We can just directly use the key in the argument here.

File: utils/db.ts [hunk 3]
```diff
@@ -208,11 +202,13 @@ export async function deleteNotification(
...
   notification: Pick<Notification, "id" | "userLogin">,
 ) {
-  return await getValue<Notification>([
+  const notificationsByIdKey = [
     "notifications_by_user",
     notification.userLogin,
     notification.id,
-  ]);
+  ];
+  const res = await kv.get<Notification>(notificationsByIdKey);
```
Comment (iuioiua) [inline utils/db.ts:210]: Ditto. Let's just use the key directly in the argument.

File: utils/db.ts [hunk 4]
```diff
@@ -481,21 +477,27 @@ export async function deleteUserBySession(sessionId: string) {
 
 /** @todo Migrate to ["users", login] key */
 export async function getUser(login: string) {
-  return await getValue<User>(["users", login]);
+  const usersByLoginKey = ["users", login];
+  const res = await kv.get<User>(usersByLoginKey);
```
Comment (iuioiua) [inline utils/db.ts:481]: Ditto

File: utils/db.ts [hunk 5]
```diff
@@ -481,21 +477,27 @@ export async function deleteUserBySession(sessionId: string) {
...
     consistency: "eventual",
-  }) ?? await getValue<User>(usersBySessionKey);
+  });
+  const res = await kv.get<User>(usersBySessionKey);
+  return eventualRes.value ?? res.value;
 }
 
 export async function getUserByStripeCustomer(stripeCustomerId: string) {
-  return await getValue<User>([
+  const usersByStripeCustomerKey = [
```
Comment (iuioiua) [inline utils/db.ts:495]: Ditto

File: utils/db.ts [hunk 6]
```diff
@@ -481,21 +477,27 @@ export async function deleteUserBySession(sessionId: string) {
...
 /** @todo Migrate to ["users", login] key */
 export async function getUser(login: string) {
-  return await getValue<User>(["users", login]);
+  const usersByLoginKey = ["users", login];
+  const res = await kv.get<User>(usersByLoginKey);
+  return res.value;
 }
 
 export async function getUserBySession(sessionId: string) {
   const usersBySessionKey = ["users_by_session", sessionId];
```
Comment (iuioiua) [inline utils/db.ts:486]:
```suggestion
  const key = ["users_by_session", sessionId];
```

File: utils/db.ts [hunk 7]
```diff
@@ -481,21 +477,27 @@ export async function deleteUserBySession(sessionId: string) {
...
-  return await getValue<User>(usersBySessionKey, {
+  const eventualRes = await kv.get<User>(usersBySessionKey, {
     consistency: "eventual",
-  }) ?? await getValue<User>(usersBySessionKey);
+  });
```
Comment (iuioiua) [inline utils/db.ts:489]:
```suggestion
  });
  if (eventualRes.value !== null) return eventualRes.value;
```

File: utils/db.ts [hunk 8]
```diff
@@ -481,21 +477,27 @@ export async function deleteUserBySession(sessionId: string) {
 
 /** @todo Migrate to ["users", login] key */
 export async function getUser(login: string) {
-  return await getValue<User>(["users", login]);
+  const usersByLoginKey = ["users", login];
+  const res = await kv.get<User>(usersByLoginKey);
+  return res.value;
 }
 
 export async function getUserBySession(sessionId: string) {
```
Comment (iuioiua) [inline utils/db.ts:485]: Previously, this function performed 1 eventual read and _possibly_ 1 consistent read. Now, with your changes, it performs both every time. Let's fix that up.

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by iuioiua — APPROVED** [review]
That's it! LGTM. Thank you, @roberto-morado.

**Pull request merged**